src/context/reducers/dicomImagesReducer.tsx

```diff
@@ -14,29 +13,90 @@ export interface DcmImage {
   StudyDescription: string,
   SeriesInstanceUID: string,
   SeriesDescription: string,
-  Modality: string
+  Modality: string,
+  pacs_identifier: string
+}
+
+export interface PACSResponseProperty {
+  tag: number;
+  value: string;
+  label: string;
+}
+
+export interface PACSResponseNumericProperty {
+  tag: number;
+  value: number;
+  label: string;
+}
+
+export interface PACSBaseResponse {
+  uid: PACSResponseNumericProperty;
+  SpecificCharacterSet: PACSResponseProperty;
+  StudyDate: PACSResponseProperty;
+  SeriesDate: PACSResponseProperty;
+  AccessionNumber: PACSResponseProperty;
+  QueryRetrieveLevel: PACSResponseProperty;
+  RetrieveAETitle: PACSResponseProperty;
+  Modality: PACSResponseProperty;
+  ModalitiesInStudy: PACSResponseProperty;
+  StudyDescription: PACSResponseProperty;
+  SeriesDescription: PACSResponseProperty;
+  PatientName: PACSResponseProperty;
+  PatientID: PACSResponseProperty;
+  PatientBirthDate: PACSResponseProperty;
+  PatientSex: PACSResponseProperty;
+  PatientAge: PACSResponseNumericProperty;
+  StudyInstanceUID: PACSResponseProperty;
+  SeriesInstanceUID: PACSResponseProperty;
+  InstanceNumber: PACSResponseProperty;
+  NumberOfSeriesRelatedInstances: PACSResponseProperty;
+  PerformedStationAETitle: PACSResponseProperty;
 }
 
-export type IDcmImagesState = DcmImage[];
+export interface PACSSeries extends PACSBaseResponse {
+  label: PACSResponseProperty;
+  command: PACSResponseProperty;
+  status: PACSResponseProperty;
+}
+
+export interface PACSMainResponse extends PACSBaseResponse {
+  series: PACSSeries[];
+}
 
-export const initialIDcmImagesState: IDcmImagesState = []
+export type IDcmImagesState = {
+  allDcmImages: DcmImage[];
+  filteredDcmImages: DcmImage[];
+}
+
+export const initialIDcmImagesState: IDcmImagesState = {
+  allDcmImages: [],
+  filteredDcmImages: []
+}
 
 type IDcmImagesStatePayload = {
-  [DicomImagesTypes.UpdateImages]: { images: DcmImage[] }
+  [DicomImagesTypes.Update_all_images]: { images: DcmImage[] },
+  [DicomImagesTypes.Update_filtered_images]: { images: DcmImage[] }
 }
 
 export type DicomImagesActions = ActionMap<IDcmImagesStatePayload>[
   keyof ActionMap<IDcmImagesStatePayload>
 ]
 
-
 export const dicomImagesReducer = (
   state: IDcmImagesState,
   action: DicomImagesActions
 ) => {
   switch (action.type) {
-    case DicomImagesTypes.UpdateImages:
-      return action.payload.images
+    case DicomImagesTypes.Update_all_images:
+      return {
+        allDcmImages: action.payload.images,
+        filteredDcmImages: action.payload.images
+      }
```

Comment:
Please, add `state` since as the state grows, there might be other attributes in addition to `allDcmImages` and `filteredDcmImages` so it's better to add `...state` to avoid future bugs

```suggestion
      return {
        ...state,
        allDcmImages: action.payload.images,
        filteredDcmImages: action.payload.images
      }
```